test/test.oauth2.ts

```diff
@@ -1348,5 +1400,58 @@ describe('oauth2', () => {
         /No refresh token is set./
       );
     });
+
+    it('should throw error is refreshHandler does not return token', async () => {
+      const expectedRefreshedAccessToken = {
+        access_token: '',
+        expiry_date: 123456789,
+      };
+      client.refreshHandlerCallback = async () => {
+        return expectedRefreshedAccessToken;
+      };
+      await assert.rejects(
+        client.refreshHandler(),
+        'There is no access token being returned.'
+      );
+    });
+
+    it('should call refreshHandler and set credential if providing refresh handler callback', async () => {
+      const expectedRefreshedAccessToken = {
+        access_token: 'access_token',
+        expiry_date: 123456789,
+      };
+      client.refreshHandlerCallback = async () => {
+        return expectedRefreshedAccessToken;
+      };
+      const refreshedAccessToken =
+        (await client.refreshHandler()) as AccessTokenResponse;
+      assert.strictEqual(
+        refreshedAccessToken.access_token,
+        expectedRefreshedAccessToken.access_token
+      );
+      assert.strictEqual(
+        refreshedAccessToken.expiry_date,
+        expectedRefreshedAccessToken.expiry_date
+      );
+    });
+
+    it('should refresh credential when refreshHandler is available in getAccessToken()', async () => {
+      const expectedRefreshedAccessToken = {
+        access_token: 'access_token',
+        expiry_date: 123456789,
+      };
+      client.refreshHandlerCallback = async () => {
+        return expectedRefreshedAccessToken;
+      };
+      client.setCredentials({
```

Comment:
Are you testing expiration time detection? If so, please update the test description:
`it('should call refreshHandler on expiration and no refresh token available'`